### PR TITLE
Re-check that we have relevant changes before looking for a release file

### PR DESCRIPTION
**Pull request by alexwlchan**
This is yet another attempt to stabilise CI with the internal_model deployment.

Here's my current best guess for what's going on, based on looking at https://buildkite.com/wellcomecollection/catalogue/builds/3342#0b14305c-7c3c-4b71-82e6-8309380ba8e2

Suppose we have the following sequence of commits:

Screenshot 2021-03-09 at 09 40 14

When you merge, we compare "merge commit" and "where you forked" – which shows up all the changes in both the yellow/square and green/triangle commits. This means that if there are changes to `internal_model` in the yellow/square commits, we'll run the `internal_model` task.

But the diff we really care about for checking a release note is between "current state of master" and "merge commit" – so let's re-check for relevant changes on the commit range we care about, before breaking a build for non-release note.

### CODE REVIEW

File: .buildkite/scripts/run_job.py
Comment (alicefuzier) [inline .buildkite/scripts/run_job.py:127]: I think I'm missing something... Why would the changed paths be different here than on line 112?

Comment (jamieparkinson) [reply .buildkite/scripts/run_job.py:127]: Agreed - I think the problem is that `commit_range` (line 83) is using the `latest` tag rather than the new `latest-sbt-release` tag (?)

Comment (jamieparkinson) [reply .buildkite/scripts/run_job.py:127]: And as we found yesterday, that tag is on the commit _before_ the version bump

Comment (alexwlchan) [reply .buildkite/scripts/run_job.py:127]: So the fix might be to compare to `latest-sbt-release` on line 83 instead?

Comment (alexwlchan) [reply .buildkite/scripts/run_job.py:127]: > Why would the changed paths be different here than on line 112?

The changed paths on line 112 are `latest..HEAD`. The changed paths here are `latest-sbt-release..HEAD`. Those two tags are set and managed differently – as Jamie says, one is before the version bump, the other is after.

(We created the `latest-sbt-release` tag yesterday to specifically point to the version bump commit.)

Comment (jamieparkinson) [reply .buildkite/scripts/run_job.py:127]: I think that's right, which makes me wonder if we do need 2 separate tags - where would we be using `latest` if that's the comparison we were making?

Comment (alexwlchan) [reply .buildkite/scripts/run_job.py:127]: We use `latest` in weco-deploy, right?

Comment (alexwlchan) [reply .buildkite/scripts/run_job.py:127]: Is `latest` something we copied across from scala-libs but doesn't make sense here sans version bump?

Comment (jamieparkinson) [reply .buildkite/scripts/run_job.py:127]: I _think_ that the `latest` in weco-deploy is an unrelated one to this git tag

Comment (alexwlchan) [reply .buildkite/scripts/run_job.py:127]: Note that even if we sort out the `latest` tag, I don't think we're out of the woods – there's about a 20–30 min delay between you merging to master and the internal_model being released/tagged, which could potentially break subsequent builds.

(I thought this was the issue with your PR yesterday.)

Comment (jamieparkinson) [reply .buildkite/scripts/run_job.py:127]: I think you're right about that, yeah (although maybe it wasn't the case for the build yesterday, I waited until everything was done to merge that PR). Trying to think of a solution for that...

Comment (alexwlchan) [reply .buildkite/scripts/run_job.py:127]: I am tempted to comment this out until we can come up with a proper fix…

Comment (alicefuzier) [reply .buildkite/scripts/run_job.py:127]: ok thanks for the explanation...I did copy across from scala-libs and yeah, I don't think the `latest` tag was ever used before that in catalogue.
I think my understanding is still a bit fuzzy, but why do we want to hold off on this fix?I would still put this solution out if it works. it's still better even if we have a problem while the internal_model is being released irghth?

Comment (alexwlchan) [reply .buildkite/scripts/run_job.py]: I am possibly a little frustrated, having fought it all of yesterday. 😓 

I've pushed a change to compare to `latest-sbt-release`, are we happy for that to go in as a stopgap?

Comment (alicefuzier) [reply .buildkite/scripts/run_job.py]: sounds ok to me. Sorry about this

**Review by alicefuzier — APPROVED** [review]

**Pull request merged**
